spec/datadog/core/telemetry/client_spec.rb

```diff
@@ -21,8 +22,8 @@
       client.worker.join
     end
 
-    context 'when no params provided' do
-      subject(:client) { described_class.new }
+    context 'with default parameters' do
+      subject(:client) { described_class.new(heartbeat_interval: heartbeat_interval) }
```

Comment:
**Note: This test was actually breaking because the let above was called `heartbeat_interval_seconds` and not `heartbeat_interval`.**

Minor: I suggest propagating the `_seconds` prefix internally as well:

```suggestion
      subject(:client) { described_class.new(heartbeat_interval_seconds: heartbeat_interval_seconds) }
```